crates/sui-types/src/transaction.rs

```diff
@@ -1010,75 +1018,139 @@ impl Display for ProgrammableMoveCall {
-                write!(f, "MoveCall({p})")
+                write!(f, "{p}")
             }
             Command::MakeMoveVec(ty_opt, elems) => {
-                write!(f, "MakeMoveVec(")?;
+                write!(f, "MakeMoveVec:\n ┌")?;
                 if let Some(ty) = ty_opt {
-                    write!(f, "Some{ty}")?;
-                } else {
-                    write!(f, "None")?;
+                    write!(f, "\n │ Type Tag: {ty}")?;
                 }
-                write!(f, ",[")?;
-                write_sep(f, elems, ",")?;
-                write!(f, "])")
+                write!(f, "\n │ Arguments:\n │   ")?;
+                write_sep(f, elems, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::TransferObjects(objs, addr) => {
-                write!(f, "TransferObjects([")?;
-                write_sep(f, objs, ",")?;
-                write!(f, "],{addr})")
+                write!(f, "TransferObjects:\n ┌\n │ Arguments: \n │   ")?;
+                write_sep(f, objs, "\n │   ")?;
+                write!(f, "\n │ Address: {addr}\n └")
             }
             Command::SplitCoins(coin, amounts) => {
-                write!(f, "SplitCoins({coin}")?;
-                write_sep(f, amounts, ",")?;
-                write!(f, ")")
+                write!(f, "SplitCoins:\n ┌\n │ Coin: {coin}\n │ Amounts: \n │   ")?;
+                write_sep(f, amounts, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::MergeCoins(target, coins) => {
-                write!(f, "MergeCoins({target},")?;
-                write_sep(f, coins, ",")?;
-                write!(f, ")")
+                write!(f, "MergeCoins:\n ┌\n │ Target: {target}\n │ Coins: \n │   ")?;
+                write_sep(f, coins, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::Publish(_bytes, deps) => {
-                write!(f, "Publish(_,")?;
-                write_sep(f, deps, ",")?;
-                write!(f, ")")
+                write!(f, "Publish:\n ┌\n │ Dependencies: \n │   ")?;
+                write_sep(f, deps, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::Upgrade(_bytes, deps, current_package_id, ticket) => {
-                write!(f, "Upgrade(_,")?;
-                write_sep(f, deps, ",")?;
-                write!(f, ", {current_package_id}")?;
-                write!(f, ", {ticket}")?;
-                write!(f, ")")
+                write!(f, "Upgrade:\n ┌\n │ Dependencies: \n │   ")?;
+                write_sep(f, deps, "\n │   ")?;
+                write!(f, "\n │ Current Package ID: {current_package_id}")?;
+                write!(f, "\n │ Ticket: {ticket}")?;
+                write!(f, "\n └")
             }
         }
     }
 }
 
 impl Display for ProgrammableTransaction {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut builder = TableBuilder::default();
+
```

Comment:
I love the way this reads in the output for the table, but I feel like this doesn't really fit with what I'd normally expect from `std::fmt::Display`. In other words, If I used display like this in an error message or something, I'd be really surprised at the output. 

I'd suggest building an entirely type. You can then maybe stick it in this file in its own section, or in a separate file. Doesn't matter much to me. But you can do

```
struct Tabled<'a, T>(&'a T)
```
and then
```
impl<'a> Display for Tabled<'a, ProgrammableTransaction>
```
And so on for the other types. This will let us have both formats: the short inline one and the longer table version. 

Curious to hear everyone's thoughts though

Reply:
That's definitely an interesting thought. The current "format" is really not readable and fairly frustrating to "parse" when you want to debug something. 

In the CLI (which uses the types from `sui-json-rpc-types`), we expect a lot of this stuff to actually be read / visualized by a human, so it makes sense to have such a readable format. For more machine-like formatting, in the CLI one can use the `--json` which will return a pretty printed JSON output. 

If I remember correctly, I think @amnn had a similar idea to yours at some point, but I did not follow up on that.

Reply:
Out of curiosity, what do you find useful for the short format?

Reply:
The current format is essentially a prettier version of a Rust debug, and it's in line with what I would "generally" expect from display, which is a short and concise representation. 

Especially here for MoveCall we are taking something like `package::module::foo<ty_args>` and splitting it over 4 rows of a table. And the table is really just not something I would really expect from `Display`.

Reply:
This is what I was specifically wondering about, I had a feeling that something like this may exist which doesn't lend well the spread-out version. 

I really like the idea of having alternate displays with `impl<'a> Display for Tabled<'a, ProgrammableTransaction>` and I think this pattern will be useful for my future work as well. this has been added now, so feel free to take a look. 

I wanted to also implement a pretty_write_sep() that wraps the iterated T with the new type, but since it takes by T reference, lifetimes didn't quite work.